evergreen/config.yml

```diff
@@ -1132,6 +1271,42 @@ buildvariants:
   - name: long-running-tests
   - name: swift-build-and-test
 
+- name: macos-coverage
+  display_name: "MacOS 11 arm64 (Code Coverage)"
+  run_on: macos-1100-arm64
+  expansions:
+    cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.20.3-macos-universal.tar.gz"
+    cmake_bindir: "./cmake_binaries/CMake.app/Contents/bin"
+    cmake_toolchain_file: "./tools/cmake/xcode.toolchain.cmake"
+    cmake_generator: Xcode
+    max_jobs: $(sysctl -n hw.logicalcpu)
+    run_tests_against_baas: On
+    xcode_developer_dir: /Applications/Xcode13.1.app/Contents/Developer
+    extra_flags: -DCMAKE_SYSTEM_NAME=Darwin -DCMAKE_OSX_ARCHITECTURES=arm64
+    enable_llvm_coverage: On
+    coveralls_flag_name: "macos-arm64"
+    grcov_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/grcov-aarch64-apple-darwin.tar.bz2"
+  tasks:
+  - name: compile_test_coverage
+
+- name: ubuntu-coverage
+  display_name: "Ubuntu (Code Coverage)"
+  run_on: ubuntu2004-large
+  expansions:
+    clang_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/clang%2Bllvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz"
+    cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.20.3-linux-x86_64.tar.gz"
+    cmake_bindir: "./cmake_binaries/bin"
+    fetch_missing_dependencies: On
+    run_tests_against_baas: On
+    c_compiler: "./clang_binaries/bin/clang"
+    cxx_compiler: "./clang_binaries/bin/clang++"
+    enable_llvm_coverage: On
+    coveralls_flag_name: "ubuntu-x86_64"
+    grcov_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/grcov-x86_64-unknown-linux-gnu.tar.bz2"
+  tasks:
+  - name: compile_test_coverage
+  - name: finalize_coverage_data
```

Comment:
Is there a race that this finalize task could complete before the macos coverage results are uploaded to coveralls?

Reply:
No, the `finalize_coverage_data` task has dependencies on all `process_coverage_data` tasks:
```
- name: finalize_coverage_data
  display_name: Finalize Coverage Data
  tags: [ "for_pull_requests" ]
  depends_on:
    - name: "process_coverage_data"
      variant: "*"
      status: "*"
```
This means that it won't run until all tasks named process_coverage_data on all variants complete with some kind of status.

Reply:
perfect, thanks for the explanation 👍